### PR TITLE
ModemController class

### PR DESCRIPTION
**Pull request by andylvua**
This update introduces a new `ModemController` class that acts as an intermediate link between the `CLI` and the `Modem`, providing a control interface. This will increase the flexibility of the project and offload the `CLI` class.

### CODE REVIEW

File: Inc/cli/cli.hpp [hunk 1]
```diff
@@ -5,42 +5,56 @@
...
 #include <ncurses.h>
 #include <unordered_map>
+#include <QSharedPointer>
+#include "screen.hpp"
+#include "cli/option.hpp"
+#include "rotary_reader/rotary_dial.hpp"
 
 using ScreenMap = QHash<QString, QSharedPointer<Screen>>;
 
+class ModemController;
```
Comment (codefloww) [inline Inc/cli/cli.hpp:21]: Is it necessary here?

Comment (codefloww) [reply Inc/cli/cli.hpp:21]: Seems that cli doesn't have modem controller

**Review by codefloww — DISMISSED** [review]
Please clean up cli.hpp and try to remove overincluding in .hpp

File: Inc/cli/cli.hpp (hunk 1 above)
Comment (andylvua) [reply Inc/cli/cli.hpp:21]: Yep, this is a necessary forward declaration here, it helps avoid circular dependencies

**Pull request merged**